packages/dds/tree/src/simple-tree/tree.ts

```diff
@@ -108,11 +106,50 @@ export interface ITreeConfigurationOptions {
 	 * - More than one MapNode type: it's impossible to tell which map type is intended in the case of an empty map (`{}`).
 	 * - Both a MapNode and an ArrayNode: this case is not a problem for the canonical JSON representation, but is an issues when constructing from an Iterable, which is supported for both MapNode and ArrayNode.
 	 * - Both a MapNode and an ObjectNode: when the input is valid for the ObjectNode, the current parser always considers it ambiguous with being a MapNode.
-	 * - ObjectNodes which don't each have a unique required field: currently every ObjectNode type in a union must have a required field with a name that only it uses for any field.
+	 * - ObjectNodes which which have fields (require or optional) which include all required fields of another ObjectNode: currently every ObjectNodes are differentiated by the presence of their required fields.
 	 *
 	 * This check is conservative: some complex cases may error if the current simple algorithm can not show no ambiguity is possible.
 	 * This check may become more permissive over time.
 	 *
+	 * @example
+	 * This example shows an ambiguous schema, and how to disambiguate it using {@link Unhydrated} nodes:
+	 * ```typescript
+	 * const schemaFactory = new SchemaFactory("com.example");
+	 * class Feet extends schemaFactory.object("Feet", { length: schemaFactory.number }) {}
+	 * class Meters extends schemaFactory.object("Meters", { length: schemaFactory.number }) {}
+	 * const config = new TreeViewConfiguration({
+	 * 	// This combination of schema is can lead to ambiguous cases, and would error if preventAmbiguity is true.
+	 * 	schema: [Feet, Meters],
+	 * 	preventAmbiguity: false,
+	 * });
+	 * const view = tree.viewWith(config);
+	 * // This is invalid since it is ambiguous which type of node is being constructed:
+	 * // view.initialize({ length: 5 });
+	 * // To work, an explicit type can be provided by using an {@link Unhydrated} Node:
+	 * view.initialize(new Meters({ length: 5 }));
+	 * ```
+	 *
+	 * @example
+	 * This example shows how to disambiguate a schema by adjusting the field names in a compatible way:
+	 * ```typescript
+	 * const schemaFactory = new SchemaFactory("com.example");
+	 * class Feet extends schemaFactory.object("Feet", { length: schemaFactory.number }) {}
+	 * class Meters extends schemaFactory.object("Meters", {
+	 * 	// To avoid ambiguity when parsing unions of Feet and Meters, this renames the length field to "meters".
+	 * 	// To preserve compatibility with existing data from the ambiguous case,
+	 * 	// `{ key: "length" }` is set, so when persisted in the tree "length" is used as the field name.
+	 * 	meters: schemaFactory.required(schemaFactory.number, { key: "length" }),
+	 * }) {}
+	 * const config = new TreeViewConfiguration({
+	 * 	// This combination of schema is can lead to ambiguous cases, and would error if preventAmbiguity is true.
```

Comment:
```suggestion
	 * 	// This combination of schema is not ambiguous because `Feet` and `Meters` have different required keys.
```

Or, just remove this comment if you think it's redundant WRT the next comment below it.